companion/lib/Data/StoreBase.ts

```diff
@@ -39,334 +48,244 @@ export class DataStoreBase {
-		} catch (err) {
-			this.logger.silly(`${this.name}_save: Error renaming ${this.name}.tmp: ` + err)
-			throw `Error renaming ${this.name}.tmp: ` + err
-		}
-
-		this.logger.silly(`${this.name}_save: renamed`)
-	}
+	protected abstract create(): void
 
 	/**
-	 * Get the entire database
-	 * @param clone - <code>true</code> if a clone is needed instead of a link
-	 * @returns the database
+	 * Close the file because we're existing
 	 */
-	getAll(clone = false): Record<string, any> {
-		let out
-		this.logger.silly(`${this.name}_all`)
-
-		if (clone === true) {
-			out = cloneDeep(this.store)
-		} else {
-			out = this.store
-		}
-
-		return out
+	public close(): void {
+		this.store?.close()
 	}
 
 	/**
-	 * @returns the directory of the flat file
+	 * Delete a key/value pair from the default table
+	 * @param key - the key to be delete
 	 */
-	getCfgDir(): string {
-		return this.cfgDir
+	public deleteKey(key: string): void {
+		this.deleteTableKey(this.defaultTable, key)
 	}
 
 	/**
-	 * @returns the flat file
+	 * Delete a key/value pair from a table
+	 * @param table - the table to delete from
+	 * @param key - the key to be delete
 	 */
-	getCfgFile(): string {
-		return this.cfgFile
+	public deleteTableKey(table: string, key: string): void {
+		if (table.length > 0 && key.length > 0) {
+			const query = this.store?.prepare(`DELETE FROM ${table} WHERE id = @id`)
+			this.logger.silly(`Delete key: ${table} - ${key}`)
+
+			try {
+				query?.run({ id: key })
+			} catch (e: any) {
+				this.logger.warn(`Error deleting ${table} - ${key}: ${e.message}`)
+			}
+
+			this.setDirty()
+		}
 	}
 
 	/**
 	 * @returns the 'is first run' flag
 	 */
-	getIsFirstRun(): boolean {
+	public getIsFirstRun(): boolean {
 		return this.isFirstRun
 	}
 
 	/**
-	 * @returns JSON of the database
+	 * Get a value from the default table
+	 * @param key - the to be retrieved
+	 * @param defaultValue  - the default value to use if the key doens't exist
+	 * @returns the value
 	 */
-	getJSON(): string | null {
-		try {
-			return JSON.stringify(this.store)
-		} catch (e) {
-			this.logger.silly(`JSON error: ${e}`)
-			return null
-		}
+	public getKey(key: string, defaultValue?: any): any {
+		return this.getTableKey(this.defaultTable, key, defaultValue)
 	}
 
 	/**
-	 * Get a value from the database
-	 * @param key - the key to be retrieved
-	 * @param defaultValue - the default value to use if the key doesn't exist
-	 * @param clone - <code>true</code> if a clone is needed instead of a link
+	 * Get all rows from a table
+	 * @param table - the table to get from
+	 * @returns the rows
 	 */
-	getKey(key: string, defaultValue?: any, clone = false): any {
-		let out
-		this.logger.silly(`${this.name}_get(${key})`)
+	public getTable(table: string): any {
+		let out = {}
 
-		if (this.store[key] === undefined && defaultValue !== undefined) {
-			this.store[key] = defaultValue
-			this.setDirty()
-		}
+		if (table.length > 0 && this.store) {
```

Comment:
should these checks cause the methods to be no-ops, or perhaps it would make more sense for them to throw an error instead? They shouldn't ever be hit anyway

Reply:
So there's a final failover to RAM and then a fatal error if it still can't do it.  Still testing through the edge cases and how the error-exit works.